File: c/stack/queue_test.c
#include <stdio.h>

#include "queue.h"
#include "../test/assert.h"

void Queue_test(int *passed, int *total) {
    printf("\n\nQueue\n\n");
    Queue q;
    Queue_init(&q);

    *passed += ASSERT_INT_EQUALS(0, Queue_size(&q), "Queue is initialized to be empty");
    *total += 1;

    Queue_enqueue(&q, 10);

    *passed += ASSERT_INT_EQUALS(1, Queue_size(&q), "Queue is correct size after first enqueue");
    *total += 1;

    Queue_enqueue(&q, 20);

    *passed += ASSERT_INT_EQUALS(2, Queue_size(&q), "Queue is correct size after second enqueue");
    *total += 1;

    int result = Queue_dequeue(&q);

    *passed += ASSERT_INT_EQUALS(10, result, "Queue dequeues correct item first");
    *total += 1;

    result = Queue_dequeue(&q);

    *passed += ASSERT_INT_EQUALS(20, result, "Queue dequeues correct item second");
    *total += 1;

    *passed += ASSERT_INT_EQUALS(0, Queue_size(&q), "Queue is empty after all items are dequeued");
    *total += 1;
}

File: Java/test/stacks_and_queues/QueueTest.java
package stacks_and_queues;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

/**
 * Created by Chris Keyser on 10/3/2016.
 */
public class QueueTest {
    private Queue<Integer> queue;

    @Before
    public void setUp() {
        queue = new Queue<>();
    }

    @Test
    public void Queue_Initialization() {
        Assert.assertTrue("Queue to be empty on initialization", queue.isEmpty());
        Assert.assertEquals("Queue to have size 0 on initialization", 0, queue.size());
    }

    @Test
    public void Queue_AfterEnqueue_HasCorrectSize() {
        queue.enqueue(5);
        queue.enqueue(11);

        Assert.assertEquals("Queue to have correct size after adding two elements", 2, queue.size());
    }

    @Test
    public void Queue_Dequeue_GetsCorrectValues() {
        queue.enqueue(5);
        queue.enqueue(11);

        Integer testVal = queue.dequeue();
        Integer expected = 5;

        Assert.assertEquals("The right element to be pulled from the queue", expected, testVal);

        testVal = queue.dequeue();
        expected = 11;

        Assert.assertEquals("The right element to be pulled from the queue", expected, testVal);
    }

    @Test
    public void Queue_Dequeue_ChangesSizeCorrectly() {
        queue.enqueue(5);
        queue.enqueue(11);
        queue.dequeue();
        queue.dequeue();

        Assert.assertEquals("The size of Queue to decrease by 1 after dequeue", 1, queue.size());
        Assert.assertEquals("The size of Queue to decrease by 1 after dequeue", 0, queue.size());
        Assert.assertTrue("The queue to be empty after all items are dequeued", queue.isEmpty());
    }
}
